packages/langium/src/lsp/language-server.ts

```diff
@@ -668,7 +672,10 @@ export function createHierarchyRequestHandler<P extends TypeHierarchySupertypesP
         }
         const language = serviceRegistry.getServices(uri);
         try {
-            return await serviceCall(language, params, cancelToken);
+            const result = await lock.read(async () => {
+                return await serviceCall(language, params, cancelToken);
+            }, true); // Give this priority, since we already waited until the target state
```

Comment:
I think this can be simplified to
```typescript
await lock.read(() => serviceCall(language, params, cancelToken), true)
```
(same in `createServerRequestHandler` and `createRequestHandler`).